### DISCUSSION TITLE
Genealogy Web Site exploiting TinyMCE

### DISCUSSION THREAD

**Discussion by jcobban** [tinymce/tinymce#8788] [Show and tell]
In the early 1990s, just after the dawn of the Web, I put my family tree up on the Web.  I chose a presentation style that resembles traditional hard-copy genealogies:

![image](https://github.com/tinymce/tinymce/assets/17891982/fa5a725f-892d-4acc-969a-5aa4e496f37b)

Indeed I largely copied the appearance of my Father's version of the family tree which he created using a word processor, except that, as you can see, I tried to use every possible opportunity to exploit the Web.  You can see the live version of this page at https://www.jamescobban.net/FamilyTree/Person.php?idir=68147&lang=en.

For a few years I edited my Family Tree on a Windows application and converted it to HTML pages using a program, but gradually I moved all of the function out to the web server.  For almost twenty years I have performed all of the family tree maintenance, as well as tracking all of my research notes and transcriptions of original documents on my web site.  The database file format I use supports a number of extended text fields, most up to 65,535 bytes, so I wanted a better editor than the default `<textarea>`.  I found TinyMCE over 15 years ago and now all of these text fields are manipulated using TinyMCE editors.  I only expose this to registered users who have permission to modify the family tree, but I do not charge for accounts.  If you want to see how this works just go to https://www.jamescobban.net/ and click on "Add a New Unrelated Individual".

![image](https://github.com/tinymce/tinymce/assets/17891982/b479b65f-9e96-49f8-9b79-d21de85fdb2f)

Note the menu displayed when you click on the "hamburger" button at the top left.  In particular note the "? Help" item which is available on all pages and displays overall guidance on the use of each page in the site.  Also note that if your prefer this particular page is also available in French.  Almost all pages on my site area available in English and French and the most popular ones, including the display of the Family Tree, are also available in Spanish, and German.

![image](https://github.com/tinymce/tinymce/assets/17891982/fb0a1c45-ed70-4ce6-9ab4-4b65cd44b660)

If you click on the "Details" button associated with any line a detail form is opened which exploits TinyMCE to support extended text fields, all of which support rich text.  Ordinary text fields only support UNICODE text entry.  Since I have been building this site for almost 30 years it is rather large, although obviously not on the scale of Ancestry.com.

**Comment by lostkeys** [comment]
@jcobban Thanks for sharing!

— Fredrik, Product Manager
